docs/src/main/sphinx/sql/set-session.rst

```diff
@@ -15,10 +15,30 @@ Description
 
 Set a session property value or a catalog session property.
 
+.. _session-properties-definition:
 
-Connectors can provide session properties. They are set separately for each
-catalog by prefixing them with the catalog name. There is no sharing of
-properties across catalogs, even if the catalogs use the same connector.
+Session properties
+------------------
+
+A session property is a :doc:`configuration property </admin/properties>` that
+can be temporarily modified by a user for the duration of the current
+connection session to the Trino cluster. Many configuration properties have a
+corresponding session property that accepts the same values as the config
+property.
+
+There are two types of session properties:
+
+* **System session properties** apply to the whole cluster. Most session
+  properties are system session properties unless specified otherwise.
+* **Catalog session properties** are connector-defined session properties that
+  can be set on a per-catalog basis. These properties can be set separately for
+  each catalog by including the catalog name as a prefix, such as
+  ``catalogname.property_name``.
+
+Session properties are tied to the current session, so a user can have multiple
+connections to a cluster that each have different values for the same session
+properties. Once a session ends, either by disconnecting or creating a new
+session, any changes made to session properties during that session are lost.
```

Comment:
nit:

```suggestion
session, any changes made to session properties during the previous session are lost.
```

Thanks, this whole paragraph reads much better now and clarifies a lot of detail.